src/main/groovy/net/vivin/gradle/versioning/SemanticBuildVersion.groovy

```diff
@@ -75,44 +48,45 @@ class SemanticBuildVersion {
             throw new BuildException("Could not autobump because there are no prior commits", e)
         }
 
-        if(lines.find { it ==~ autobumpConfiguration.newPreReleasePattern}) {
-            newPreRelease = true
-        }
-
-        if(lines.find { it ==~ autobumpConfiguration.majorPattern }) {
-            bump = VersionComponent.MAJOR
-        } else if(lines.find { it ==~  autobumpConfiguration.minorPattern }) {
-            bump = VersionComponent.MINOR
-        } else if(lines.find { it ==~ autobumpConfiguration.patchPattern }) {
-            bump = VersionComponent.PATCH
-        } else if(lines.find { it ==~ autobumpConfiguration.preReleasePattern }) {
-            if(newPreRelease) {
-                throw new BuildException("Could not autobump because it is not possible to bump the pre-release version when also creating a new pre-release version", null)
-            }
-
-            bump = VersionComponent.PRERELEASE
-        } else if(lines.find { it ==~ autobumpConfiguration.promoteToReleasePattern }) {
-            if(newPreRelease) {
-                throw new BuildException("Could not autobump because it is not possible to promote to a release version when also creating a new pre-release version", null)
-            }
-
-            promoteToRelease = true
-        } else {
-            if(!newPreRelease) {
-                throw new BuildException("Could not autobump because the last commit message did not match the major (/${autobumpConfiguration.majorPattern}/), minor (/${autobumpConfiguration.minorPattern}/), patch (/${autobumpConfiguration.patchPattern}/), pre-release (/${autobumpConfiguration.preReleasePattern}/), or release (/${autobumpConfiguration.promoteToReleasePattern}/) patterns specified in the autobump configuration", null)
-            }
-
-            bump = VersionComponent.PATCH
+        newPreRelease = lines.any { it ==~ config.autobump.newPreReleasePattern }
+
+        switch(lines) {
+            case { it.any { it ==~ config.autobump.majorPattern } }:
```

Comment:
Yep, switch-cases are cool in Groovy.
You can put anything there that has an `isCase(Object)` method and as Groovy provides this method for `Object`, you can put anything there and it will be compared with `equals()` unless the class overwrites `isCase(Object)`. e. g. for arrays and collections, it is checked whether the switch-value is contained, for `Pattern` a regex match is done, for closures it is checked whether they return `true` with the switch-value as argument and for a class an `instanceof` check is done.